Unable to run 'make tests' to gather unit test results

**Describe the bug**
This repository should allow running a suite of unit tests for the component.  Currently this doesn't appear to be working.

In particular, when I attempt this locally using NodeJS v18, the following output (lightly edited to remove local file paths) is produced:

```
frontend$ make tests
yarnpkg run mocha --mode 'development' --require setup.mjs --require ts-node/register 'src/**/*.spec.ts'
yarn run v1.22.19
$ node_modules/.bin/mocha --mode development --require setup.mjs --require ts-node/register 'src/**/*.spec.ts'

TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".ts" for src/app/state.spec.ts
    at new NodeError (node:internal/errors:400:5)
    at Object.getFileProtocolModuleFormat [as file:] (node:internal/modules/esm/get_format:79:11)
    at defaultGetFormat (node:internal/modules/esm/get_format:121:38)
    at defaultLoad (node:internal/modules/esm/load:81:20)
    at nextLoad (node:internal/modules/esm/loader:163:28)
    at ESMLoader.load (node:internal/modules/esm/loader:605:26)
    at ESMLoader.moduleProvider (node:internal/modules/esm/loader:457:22)
    at new ModuleJob (node:internal/modules/esm/module_job:64:26)
    at ESMLoader.#createModuleJob (node:internal/modules/esm/loader:480:17)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:434:34)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
make: *** [Makefile:46: tests] Error 1
```

Some standardisation of the NodeJS, Yarn, and/or other component versions may be required to resolve the problem.

**To Reproduce**
Steps to reproduce the behavior:
1. From the commandline, run `make tests`.

**Expected behavior**
The unit test suite should complete and `mocha` should emit success/failure statistics for the test run.

--- a/src/app/main.ts
+++ b/src/app/main.ts
@@ -1,5 +1,3 @@
-import 'bootstrap/dist/css/bootstrap.css';
-import 'bootstrap-table/dist/bootstrap-table.css';
 import 'select2/dist/css/select2.css';
 import 'tablesaw/dist/stackonly/tablesaw.stackonly.css';
 
@@ -20,6 +18,10 @@ import './views/components/recipe-list.css';
 import './main.css';
 
 import './database';
+
+import 'bootstrap-table';
+import 'bootstrap/dist/css/bootstrap.css';
+import 'bootstrap-table/dist/bootstrap-table.css';
 
 import { recipeFormatter, rowAttributes } from './views/components/recipe-list';
 export { recipeFormatter, rowAttributes };

--- a/src/app/views/components/recipe-list.ts
+++ b/src/app/views/components/recipe-list.ts
@@ -20,7 +20,7 @@ export {
     updateRecipeState,
 };
 
-function attributionFormatter(recipe: Recipe) : JQuery {
+function attributionFormatter(recipe: Recipe) : $ {
   const container = $('<div />', {'class': 'attribution'});
 
   const title = $('<a />', {
@@ -49,7 +49,7 @@ function starFormatter() {
   return $('<div />', {'class': 'star', 'html': '&#x269d;'});
 }
 
-function thumbnailFormatter(recipe) : JQuery {
+function thumbnailFormatter(recipe) : $ {
   const container = $('<td />', {'class': 'thumbnail align-top'});
   const link = $('<a />', {'href': recipe.dst});
   const img = $('<img />', {
@@ -62,7 +62,7 @@ function thumbnailFormatter(recipe) : JQuery {
   return container;
 }
 
-function sidebarFormatter(recipe) : JQuery {
+function sidebarFormatter(recipe) : $ {
   const duration = Duration.fromObject({minutes: recipe.time}, {locale: resolvedLocale()});
 
   const sidebar = $('<td />', {'class': 'sidebar align-top'});

--- a/src/app/views/components/recipe-list.ts
+++ b/src/app/views/components/recipe-list.ts
@@ -236,7 +236,9 @@ function bindPostBody(selector: string) : void {
     $(this).find('button.add-recipe').each((_, button) => {
       $(button).on('click', () => { getRecipe(button).then(addRecipe).then(updateRecipeState); });
     });
-    $(this).parents('div.recipe-list').show();
+    if (data && data.length) {
+      $(this).parents('div.recipe-list').show();
+    }
 
     // Localize search result elements
     localize(selector);

--- a/src/app/views/search.ts
+++ b/src/app/views/search.ts
@@ -1,5 +1,4 @@
 import * as $ from 'jquery';
-import 'bootstrap-table';
 import { debounce } from 'debounce';
 
 import '../autosuggest';
@@ -74,7 +73,7 @@ function renderSearch() : void {
   });
 }
 
-function renderRefinement(refinement: string) : JQuery {
+function renderRefinement(refinement: string) : $ {
   if (refinement == 'empty_query') {
     return $('<div />', {
       'data-i18n': i18nAttr('search:refinement-empty-query')
@@ -104,7 +103,7 @@ function updateStateDomains() : void {
     debouncedSearchTrigger();
 }
 
-function renderDomainFacet(domain: Record<string, string>, state?: boolean) : JQuery {
+function renderDomainFacet(domain: Record<string, string>, state?: boolean) : $ {
   const domainState = state === undefined ? true : state;
   const chip = $('<label />', {'class': 'badge bg-light rounded-pill text-dark'});
   const checkbox = $('<input />', {'type': 'checkbox', 'checked': domainState, 'value': domain.key});

